### PR TITLE
modal update

### PR DESCRIPTION
**Pull request by kidkai25**
used Material UI Modal Component
old component is also there

### CODE REVIEW

**Comment by lets-learn-react** [comment]
I pulled this branch and the other branch `kidkai` - the other branch had a modal but this one I couldn't get t to fire. Which one should I review?

**Comment by kidkai25** [comment]
> I pulled this branch and the other branch kidkai - the other branch had a modal but this one I couldn't get t to fire. Which one should I review?

sorry I had not added the updated card component. Made a new pr, should fire now

File: src/components/CardComponent.js [hunk 1]
```diff
@@ -6,34 +6,61 @@ import CardContent from '@material-ui/core/CardContent'
...
-        <Button size='small' color='primary'>
-          View
+        </CardContent>
+        <CardActions>
+          <Button size='small' color='primary' onClick={() => this.openJobModal()}>
+            View
         </Button>
-        <Button size='small' color='primary'>
-          Edit
+          {/* <JobModal isOpen={this.state.isModalOpen} onClose={() => this.closeJobModal()}></JobModal> */}
```
Comment (lets-learn-react) [inline src/components/CardComponent.js]: remove commented out code

File: src/components/CardComponent.js [hunk 2]
```diff
@@ -6,34 +6,61 @@ import CardContent from '@material-ui/core/CardContent'
...
 
+class CardComponent extends React.Component {
+
+
+  constructor(props) {
+    super(props)
+    this.state = { isModalOpen: false }
+  }
+
+  openJobModal() {
```
Comment (lets-learn-react) [inline src/components/CardComponent.js]: you don't need both of these - you can just do `toggleJobModal () { this.setState({ isOpen: !this.state.isOpen }`

File: src/components/CardComponent.js [hunk 3]
```diff
@@ -6,34 +6,61 @@ import CardContent from '@material-ui/core/CardContent'
...
-          View
+        </CardContent>
+        <CardActions>
+          <Button size='small' color='primary' onClick={() => this.openJobModal()}>
+            View
         </Button>
-        <Button size='small' color='primary'>
-          Edit
+          {/* <JobModal isOpen={this.state.isModalOpen} onClose={() => this.closeJobModal()}></JobModal> */}
+          <JobModalNew isOpen={this.state.isModalOpen} onClose={() => this.closeJobModal()}></JobModalNew>
```
Comment (lets-learn-react) [inline src/components/CardComponent.js]: you're `calling` this function `onClose` and then defining `onClose` as a separate function inside `JobModalNew` - that doesn't really make sense.  You should pass the new `toggleModal` method as `onClose` prop -- pass, not call.  Then in the JobModalNew component you can call the method on close.

File: src/components/JobModal.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React from 'react'
```
Comment (lets-learn-react) [inline src/components/JobModal.js]: Remove this component if it's not being used.

File: src/components/JobModalFields.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+
+export default (props) => {
+
+    let styleRight = {
```
Comment (lets-learn-react) [inline src/components/JobModalFields.js]: define. the style object _outside_ of the compenent

File: src/components/JobModalFields.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+
+export default (props) => {
```
Comment (lets-learn-react) [inline src/components/JobModalFields.js]: don't pass `props` - destructure the individual props that you're using so we have more control and we're not hiding props

File: src/components/JobModalFields.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    };
+    let container = {
+        marginLeft: "10px",
+        marginRight: "10px",
+        marginTop: "10%",
+        marginBottom: "10%"
+    }
+    return (
+      
+        <div style={{ top: '50%', left: '50%', transform: `translate(-50%,-50%)` }} className={props.className}>
```
Comment (lets-learn-react) [inline src/components/JobModalFields.js]: here we're using `inline styles`, i.e. `{{ top: '50%'` and mixing them with `CSS-In-JS` above.  We shouldn't do that - one or the other

File: src/components/JobModalNew.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        position: 'absolute',
+        width: theme.spacing.unit * 50,
+        backgroundColor: theme.palette.background.paper,
+        boxShadow: theme.shadows[5],
+        padding: theme.spacing.unit * 4,
+        outline: 'none',
+    },
+});
+
+class JobModalNew extends React.Component {
```
Comment (lets-learn-react) [inline src/components/JobModalNew.js]: this doesn't need to be a class and we don't need the `handleClose` method as we're defining similar in the `CardComponent`

File: src/components/JobModalNew.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    render() {
+        const { classes } = this.props;
+
+        return (
+            <Modal
+                aria-labelledby="Job description"
+                aria-describedby="This is Job Modal"
+                open={this.props.isOpen}
```
Comment (lets-learn-react) [inline src/components/JobModalNew.js]: we're pulling `classes` off props above but not `isOpen`.  Why?

**Review by lets-learn-react — CHANGES_REQUESTED** [review]
Some thoughts and comments.  I'd really like to keep consistent styling so please install `standardJS` as your linter and abide by those linting rules.  it'll help catch errors and bugs too!

File: src/components/CardComponent.js (hunk 3 above)
Comment (kidkai25) [reply src/components/CardComponent.js]: Just to inform, 'onClose' is a prop of Modal api that expects a callback function that is fired when the component is closed.
https://material-ui.com/api/modal/

**Review by lets-learn-react — APPROVED** [review]
Looks good!  We'll have to work the design inside the modal, but this works! Go ahead and merge

**Pull request merged**